### ISSUE TITLE
p2 to p3 invalid scope

### ISSUE THREAD

**Issue by RonaldRizkDev** [IdentityServer/IdentityServer4#4246]
When i upgrade from preview 2 to preview 3 my scope becomes invalid.  if i rollback it works just fine.  what was changed in here?

**Comment by brockallen** [comment]
lots of redesign in that area. there will be breaking changes (including DB schema) in v4.0.

**Comment by RonaldRizkDev** [comment]
Are there notes on how to fix it?  we are already hobbled together because we are using username/password as the validation.  Our grant is ResourceOwnerPassword and we needed to write  our own ValidateAsync because that grant is deprecated.  How do I work around this issue?  I have 4 sites waiting to be upgraded.  Thanks for your assistance

**Comment by brockallen** [comment]
For now, no -- we just released the preview in the past week. I'd suggest looking at the new config in our dev host: https://github.com/IdentityServer/IdentityServer4/tree/master/src/IdentityServer4/host/Configuration

If you were on preview 2, then you're certainly not in production yet. Just delete the DB and start over.

**Comment by RonaldRizkDev** [comment]
No, I am in production.  I have been using ID4 for a year and a half, when preview 3 didn’t work I just took one step back.  I could just as easily have stepped back to 3.1.2.  so let me rephrase the question.  Since 4.x has breaking changes, I cannot obviously update any further at the moment, which is fine, there aren’t any new features I am dying for at the moment, we only use the most rudimentary auth/auth functionality anyways.  Will there be a migration path from 3.x to 4.x short of deleting all my existing users?

**Comment by brockallen** [comment]
> Will there be a migration path from 3.x to 4.x short of deleting all my existing users?

The user DB is not what I was talking about. I was talking about the IdentityServer DB: https://identityserver4.readthedocs.io/en/latest/reference/ef.html

IdentityServer has never had a mandated user DB, and that's always been the responsibility of your hosting app (meaning you and your code).
